### PR TITLE
fix(TUP-19694):testcase version changed when open job as another version

### PR DESCRIPTION
**Pull request by hwang-talend**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/preference/VersionManagementPage.java [hunk 1]
```diff
@@ -382,6 +385,33 @@ private void selectAllDependencies() {
                         ExceptionHandler.process(et);
                     }
                 }
+                
+                if(object.getItem() instanceof ProcessItem){
+                	RepositoryNode node = object.getRepositoryNode();
+                	IContentProvider contentProvider = treeViewer.getContentProvider();
+                    if (contentProvider instanceof NavigatorContentServiceContentProvider) {
+                        NavigatorContentServiceContentProvider navigatorProvider = (NavigatorContentServiceContentProvider) contentProvider;
+                        Object[] children = navigatorProvider.getChildren(node);
```
Comment (ggu-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/preference/VersionManagementPage.java]: Not sure it's safe later, if we change the CNF, maybe should try the safe way:
like, getAllTestContainers methord?

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/preference/AbstractVersionManagementProjectSettingPage.java [hunk 2]
```diff
@@ -483,6 +484,23 @@ protected void processItems(List<ItemVersionObject> objects, RepositoryNode node
                     ItemVersionObject object = createItemVersionObject(node, property);
                     objects.add(object);
                 }
+                if(item instanceof ProcessItem){
+                	IContentProvider contentProvider = treeViewer.getContentProvider();
+                    if (contentProvider instanceof NavigatorContentServiceContentProvider) {
+                        NavigatorContentServiceContentProvider navigatorProvider = (NavigatorContentServiceContentProvider) contentProvider;
+                        Object[] children = navigatorProvider.getChildren(node);
```
Comment (ggu-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/preference/AbstractVersionManagementProjectSettingPage.java]: Not sure it's safe later, if we change the CNF, maybe should try the safe way:
like, getAllTestContainers methord?

File: main/plugins/org.talend.repository/src/main/java/org/talend/repository/preference/VersionManagementPage.java (hunk 1 above)
Comment (ggu-talend) [inline main/plugins/org.talend.repository/src/main/java/org/talend/repository/preference/VersionManagementPage.java]: Same

File: main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/wizards/OpenExistVersionProcessWizard.java [hunk 3]
```diff
@@ -283,10 +283,10 @@ public int compare(ProcessItem o1, ProcessItem o2) {
...
         try {
             repositoryFactory.save(getProperty(), this.originaleObjectLabel, this.originalVersion);
-            for (ProcessItem item : testcaseList) {
-            	item.getProperty().setVersion(mainPage.getNewVersion());
-            	repositoryFactory.save(ProjectManager.getInstance().getCurrentProject(), item.getProperty());
-			}
+//            for (ProcessItem item : testcaseList) {
+//            	item.getProperty().setVersion(mainPage.getNewVersion());
+//            	repositoryFactory.save(ProjectManager.getInstance().getCurrentProject(), item.getProperty());
+//			}
```
Comment (nrousseau) [inline main/plugins/org.talend.designer.core/src/main/java/org/talend/designer/core/ui/wizards/OpenExistVersionProcessWizard.java]: if it's really not used anymore, code needs to be removed, not commented !

**Pull request merged**